src/support-prompt-group/__tests__/support-prompt-group.test.tsx

```diff
@@ -0,0 +1,164 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { cleanup, fireEvent, render } from "@testing-library/react";
+import { afterEach, describe, expect, test, vi } from "vitest";
+
+import { KeyCode } from "@cloudscape-design/component-toolkit/internal";
+
+import SupportPromptGroup, { SupportPromptGroupProps } from "../../../lib/components/support-prompt-group";
+import createWrapper from "../../../lib/components/test-utils/dom";
+
+import styles from "../../../lib/components/support-prompt-group/styles.selectors.js";
+
+const onItemClick = vi.fn();
+
+const defaultProps = {
+  onItemClick,
+  ariaLabel: "Test support prompt group",
+  items: [
+    {
+      text: "Item 1",
+      id: "item-1",
+    },
+    {
+      text: "Item 2",
+      id: "item-2",
+    },
+    {
+      text: "Item 3",
+      id: "item-3",
+    },
+  ],
+};
+
+export function renderSupportPromptGroup(
+  props: Partial<SupportPromptGroupProps>,
+  ref?: React.Ref<SupportPromptGroupProps.Ref>,
+) {
+  const renderResult = render(<SupportPromptGroup ref={ref} {...defaultProps} {...props} />);
+  const wrapper = createWrapper(renderResult.container).findSupportPromptGroup()!;
+
+  return wrapper;
+}
+
+describe("Support prompt group", () => {
+  test("Renders null with no items", () => {
+    const wrapper = renderSupportPromptGroup({
+      items: [],
+    });
+
+    expect(wrapper).toBeNull();
+  });
+
+  test("Finds number of items", () => {
+    const wrapper = renderSupportPromptGroup({});
+
+    expect(wrapper.findItems().length).toBe(3);
+  });
+
+  test("Finds item by id", () => {
+    const wrapper = renderSupportPromptGroup({});
+
+    expect(wrapper.findItemById("item-1")?.getElement()).toHaveTextContent("Item 1");
```

Comment:
Not critical, but we can use `!` here instead of `?`, since we expect that element to exist. This way we can catch the error slightly earlier and thus have a bit clearer error message